ydb/core/kqp/session_actor/kqp_query_state.h

```diff
@@ -367,10 +373,20 @@ class TKqpQueryState : public TNonCopyable {
                 }
             }
         }
+        TxCtx->HasImmediateEffects |= tx && tx->GetHasEffects();
```

Comment:
this line means that we have sink effect, isn't it?

Reply:
Any effect. In case of datashard it's upsert's/delete's/..., in case of columnshard it's sinks.